app/models/katello/repository.rb

```diff
@@ -276,29 +276,34 @@ def product_type
       redhat? ? "redhat" : "custom"
     end
 
-    def self.errata_with_package_counts(repo)
+    def self.errata_with_package_counts(repo, repository_mapping = {})
       repository_rpm = Katello::RepositoryRpm.table_name
       repository_errata = Katello::RepositoryErratum.table_name
       rpm = Katello::Rpm.table_name
       errata = Katello::Erratum.table_name
       erratum_package = Katello::ErratumPackage.table_name
+      repositories_to_search = [repo.id]
+      repositories_to_search += Repository.where(:pulp_id => repository_mapping.keys).pluck(:id) unless repository_mapping.blank?
+      where_clause = repositories_to_search.map do |repo_id|
+        "(#{repository_rpm}.repository_id = #{repo_id} and #{repository_errata}.repository_id = #{repo_id})"
+      end
+
       ::Katello::Erratum.joins(
         "INNER JOIN #{erratum_package} on #{erratum_package}.erratum_id = #{errata}.id",
         "INNER JOIN #{repository_errata} on #{repository_errata}.erratum_id = #{errata}.id",
         "INNER JOIN #{rpm} on #{rpm}.filename = #{erratum_package}.filename",
         "INNER JOIN #{repository_rpm} on #{repository_rpm}.rpm_id = #{rpm}.id").
-        where("#{repository_rpm}.repository_id" => repo.id).
-        where("#{repository_errata}.repository_id" => repo.id).
-        group("#{errata}.id").count
+         where(where_clause.join(" OR ")).
+         group("#{errata}.id").count
     end
 
-    def partial_errata
+    def partial_errata(repository_mapping = {})
```

Comment:
why exactly is repository_mapping needed to figure out partially included errata in a repository?